WordPress/Classes/ViewRelated/Reader/ReaderDetailViewController.swift

```diff
@@ -0,0 +1,903 @@
+            return
+        }
+        previewSite()
+    }
+
+
+    @IBAction func didTapBlogNameButton(sender: UIButton) {
+        previewSite()
+    }
+
+
+    @IBAction func didTapMenuButton(sender: UIButton) {
+        ReaderPostMenu.showMenuForPost(post!, fromView: menuButton, inViewController: self)
+    }
+
+
+    func didTapFeaturedImage(gesture: UITapGestureRecognizer) {
+        if gesture.state != .Ended {
+            return
+        }
+
+        let controller = WPImageViewController(image: featuredImageView.image)
+        controller.modalTransitionStyle = .CrossDissolve
+        controller.modalPresentationStyle = .FullScreen
+        presentViewController(controller, animated: true, completion: nil)
+    }
+
+
+    func didTapDiscoverAttribution() {
+        if post?.sourceAttribution == nil {
+            return
+        }
+
+        if let blogID = post?.sourceAttribution.blogID {
+            let controller = ReaderStreamViewController.controllerWithSiteID(blogID, isFeed: false)
+            navigationController?.pushViewController(controller, animated: true)
+            return
+        }
+
+        var path: String?
+        if post?.sourceAttribution.attributionType == SourcePostAttributionTypePost {
+            path = post?.sourceAttribution.permalink
+        } else {
+            path = post?.sourceAttribution.blogURL
+        }
+
+        if let linkURL = NSURL(string: path!) {
+            presentWebViewControllerWithURL(linkURL)
+        }
+    }
+
+
+    func didTapShareButton(sender: UIButton) {
+        ReaderHelpers.sharePost(post!, fromView: sender, inViewController: self)
+    }
+
+
+    func handleBlockSiteNotification(notification:NSNotification) {
+        if let userInfo = notification.userInfo, aPost = userInfo["post"] as? NSObject {
+            if aPost == post! {
+                navigationController?.popViewControllerAnimated(true)
+            }
+        }
+    }
+}
+
+
+// MARK: - WPRichTextView Delegate Methods
+
+extension ReaderDetailViewController : WPRichTextViewDelegate
+{
+
+    public func richTextView(richTextView: WPRichTextView!, didReceiveImageLinkAction imageControl: WPRichTextImage!) {
+        var controller: WPImageViewController
+
+        if WPImageViewController.isUrlSupported(imageControl.linkURL) {
+            controller = WPImageViewController(image: imageControl.imageView.image, andURL: imageControl.linkURL)
+
+        } else if let linkURL = imageControl.linkURL {
+            presentWebViewControllerWithURL(linkURL)
+            return
+
+        } else {
+            controller = WPImageViewController(image: imageControl.imageView.image)
+        }
+
+        controller.modalTransitionStyle = .CrossDissolve
+        controller.modalPresentationStyle = .FullScreen
+
+        presentViewController(controller, animated: true, completion: nil)
+    }
+
+
+    public func richTextView(richTextView: WPRichTextView!, didReceiveLinkAction linkURL: NSURL!) {
```

Comment:
If linkURL can never be nil, can we kill the `!` ?. (If it can, line 876 will explode)